packages/commerce-sdk-react/src/hooks/useQuery.ts

```diff
@@ -66,3 +70,64 @@ export const useQuery = <Client extends ApiClient, Options extends ApiOptions, D
             : {})
     })
 }
+
+/**
+ * A hook for SCAPI custom endpoint queries.
+ *
+ * Besides calling custom endpoint, this hook does a few things for better DX.
+ * 1. inject access token
+ * 2. merge SCAPI client configurations from the CommerceApiProvider
+ * @param apiOptions - Options passed through to commerce-sdk-isomorphic
+ * @param queryOptions - Options passed through to @tanstack/react-query
+ * @returns A TanStack Query query hook with data from the custom API endpoint.
+ */
+export const useCustomQuery = (
+    apiOptions: CustomEndpointArgClientConfigOptional,
+    queryOptions?: UseQueryOptions<unknown, unknown, unknown, any>
+) => {
+    const config = useConfig()
+    const auth = useAuthContext()
+    const callCustomEndpointWithAuth = (options: CustomEndpointArgClientConfigOptional) => {
+        const clientConfig = options.clientConfig || {}
+        const clientHeaders = config.headers || {}
+        return async () => {
+            const {access_token} = await auth.ready()
+            return await helpers.callCustomEndpoint({
+                ...options,
+                options: {
+                    ...options.options,
+                    method: options.options.method || 'GET',
+                    headers: {
+                        Authorization: `Bearer ${access_token}`,
+                        ...clientHeaders,
+                        ...options.options?.headers
+                    }
+                },
+                clientConfig: {
+                    parameters: {
+                        clientId: config.clientId,
+                        siteId: config.siteId,
+                        organizationId: config.organizationId,
+                        shortCode: config.organizationId
+                    },
+                    proxy: config.proxy,
+                    ...clientConfig
+                }
+            })
+        }
+    }
+
+    // Following the query key convention in this repo, the first element of the query key is a static prefix
+    // the following elements are the path components of the endpoint
+    const queryKey = [
+        '/commerce-sdk-react',
+        '/custom',
+        `/${apiOptions.options.customApiPathParameters?.apiName || 'apiName'}`,
+        `/${apiOptions.options.customApiPathParameters?.apiVersion || 'apiVersion'}`,
+        `/organizations`,
+        `/${apiOptions.options.customApiPathParameters?.organizationId || 'organizationId'}`,
+        `/${apiOptions.options.customApiPathParameters?.endpointPath || 'endpointPath'}`,
```

Comment:
I'm understanding the default values used to fallback on here, e.g. the static value 'apiName',  'apiVersion', 'organizationId' and 'endpointPath'.

1. If apiName isn't defined we probably already failed making the call.
2. we'll probably have to have duplicate logic that defines what the apiVersion is.. alternatively we have it exported from the isomorphic lib.

Reply:
i think `customApiPathParameters` should be required, but it's optional at the moment, shall we fix that in commerce-sdk-isomorphic

Reply:
Good call. i added a guard to make sure customApiPathParameters is defined before calling the api. otherwise throw error